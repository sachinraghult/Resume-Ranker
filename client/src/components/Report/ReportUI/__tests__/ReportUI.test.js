import React from "react";
global.React = React;
import ReactDOM from "react-dom/client";
import Bargraph from "../Bargraph";
import Heatmap from "../Heatmap";
import ReportUI from "../ReportUI";
import Scattercharts from "../Scattercharts";
import Table from "../Table";

import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { BrowserRouter } from "react-router-dom";

it("renders without crashing", async () => {
  const root = ReactDOM.createRoot(document.createElement("div"));
  await act(async () =>
    render(
      <BrowserRouter>
        <Bargraph />
        <Heatmap />
        <ReportUI />
        <Scattercharts />
        <Table />
      </BrowserRouter>
    )
  );
});

it("renders correctly", () => {
  const bd = {
    post: {
      _id: "62c9ebadf448b0664028a3d8",
      user: "62c9ea1df448b0664028a3c2",
      title: "Software Engineer",
      desc: "Check out latest 2020 Wells Fargo job vacancies in India. Get details on salary, company and location. Apply quickly to various Wells Fargo jobs ",
      exp: "Bachelor's degree in Engineering. \\nGood spoken and written English skills to effectively communicate technical concepts. \\nStrong logical, analytical skills and a systematic problem solving approach. \\nGood understanding of Software Development Life Cycle (SDLC).\\nDetail oriented, results driven, accountable and ability to work with multiple priorities.\\nAbility to research and report on a variety of issues using problem solving skills.\\nAbility to interact with integrity and a high level of professionalism with all levels of team members and management.\\nAbility to make timely and independent judgment decisions while working in a fast paced and results-driven environment",
      skills: [
        {
          skill: "Java",
          value: 40,
          _id: "62c9ebadf448b0664028a3d9",
        },
        {
          skill: "C#",
          value: 10,
          _id: "62c9ebadf448b0664028a3da",
        },
        {
          skill: "SQL",
          value: 15,
          _id: "62c9ebadf448b0664028a3db",
        },
        {
          skill: "Python",
          value: 35,
          _id: "62c9ebadf448b0664028a3dc",
        },
      ],
      tags: ["Internship", "Full Time"],
      deadline: "2022-07-31T00:00:00.000Z",
      email: "wellsfargo@gmail.com",
      applications: [
        "62cb1d52ab7429104fb50d14",
        "62cb1de2ab7429104fb50d7f",
        "62cb1e7aab7429104fb50e08",
      ],
      createdAt: "2022-07-09T20:57:17.768Z",
      updatedAt: "2022-07-10T18:46:18.015Z",
      __v: 12,
      keywords: [],
    },
    details: [
      {
        users: "User 3",
        scores: [
          0.7682849284825344, 0.4470739270719687, 0.723619182621134, 0.1, 0.1,
          0.1, 0.1,
        ],
      },
      {
        users: "User1",
        scores: [
          0.6473675046230588, 0.16339075919433416, 0.7211929349627837, 0.1, 0.1,
          0.1, 0.1,
        ],
      },
      {
        users: "User 2",
        scores: [
          0.02574611905749618, 0.06143623032533683, 0.03070897970876288, 0.1,
          0.1, 0.1, 0.1,
        ],
      },
    ],
  };

  const sc = {
    post: {
      _id: "62c9ebadf448b0664028a3d8",
      user: "62c9ea1df448b0664028a3c2",
      title: "Software Engineer",
      desc: "Check out latest 2020 Wells Fargo job vacancies in India. Get details on salary, company and location. Apply quickly to various Wells Fargo jobs ",
      exp: "Bachelor's degree in Engineering. \\nGood spoken and written English skills to effectively communicate technical concepts. \\nStrong logical, analytical skills and a systematic problem solving approach. \\nGood understanding of Software Development Life Cycle (SDLC).\\nDetail oriented, results driven, accountable and ability to work with multiple priorities.\\nAbility to research and report on a variety of issues using problem solving skills.\\nAbility to interact with integrity and a high level of professionalism with all levels of team members and management.\\nAbility to make timely and independent judgment decisions while working in a fast paced and results-driven environment",
      skills: [
        {
          skill: "Java",
          value: 40,
          _id: "62c9ebadf448b0664028a3d9",
        },
        {
          skill: "C#",
          value: 10,
          _id: "62c9ebadf448b0664028a3da",
        },
        {
          skill: "SQL",
          value: 15,
          _id: "62c9ebadf448b0664028a3db",
        },
        {
          skill: "Python",
          value: 35,
          _id: "62c9ebadf448b0664028a3dc",
        },
      ],
      tags: ["Internship", "Full Time"],
      deadline: "2022-07-31T00:00:00.000Z",
      email: "wellsfargo@gmail.com",
      applications: [
        "62cb1d52ab7429104fb50d14",
        "62cb1de2ab7429104fb50d7f",
        "62cb1e7aab7429104fb50e08",
      ],
      createdAt: "2022-07-09T20:57:17.768Z",
      updatedAt: "2022-07-10T18:46:18.015Z",
      __v: 12,
      keywords: [],
    },
    details: [
      {
        scores: [
          0.02574611905749618, 0.06143623032533683, 0.03070897970876288, 0.1,
          0.1, 0.1, 0.1,
        ],
      },
      {
        scores: [
          0.6473675046230588, 0.16339075919433416, 0.7211929349627837, 0.1, 0.1,
          0.1, 0.1,
        ],
      },
      {
        scores: [
          0.7682849284825344, 0.4470739270719687, 0.723619182621134, 0.1, 0.1,
          0.1, 0.1,
        ],
      },
    ],
  };
  render(
    <BrowserRouter>
      <Bargraph bargraph={bd} />
      <Heatmap />
      <ReportUI />
      <Scattercharts scatter={sc} />
      <Table tableDetails={bd} />
    </BrowserRouter>
  );
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Bargraph />
        <Heatmap />
        <ReportUI />
        <Scattercharts />
        <Table />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
