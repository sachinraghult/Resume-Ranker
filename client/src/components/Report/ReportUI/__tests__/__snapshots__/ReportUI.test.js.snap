// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    height="100%"
    style={
      Object {
        "marginBottom": "50px",
        "marginTop": "25px",
      }
    }
    width="100%"
  >
    <h3
      style={
        Object {
          "marginBottom": "25px",
        }
      }
    >
      Distribution of Top Applicants Score
    </h3>
    <div
      className="recharts-wrapper"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "default",
          "height": 300,
          "position": "relative",
          "width": 500,
        }
      }
    >
      <svg
        className="recharts-surface"
        height={300}
        version="1.1"
        viewBox="0 0 500 300"
        width={500}
      >
        <defs>
          <clipPath
            id="recharts19-clip"
          >
            <rect
              height={235}
              width={390}
              x={80}
              y={30}
            />
          </clipPath>
        </defs>
        <g
          className="recharts-cartesian-grid"
        >
          <g
            className="recharts-cartesian-grid-horizontal"
          >
            <line
              fill="none"
              height={235}
              offset={
                Object {
                  "bottom": 35,
                  "brushBottom": 35,
                  "height": 235,
                  "left": 80,
                  "right": 30,
                  "top": 30,
                  "width": 390,
                }
              }
              stroke="#ccc"
              strokeDasharray="3 3"
              width={390}
              x={80}
              x1={80}
              x2={470}
              y={30}
              y1={265}
              y2={265}
            />
            <line
              fill="none"
              height={235}
              offset={
                Object {
                  "bottom": 35,
                  "brushBottom": 35,
                  "height": 235,
                  "left": 80,
                  "right": 30,
                  "top": 30,
                  "width": 390,
                }
              }
              stroke="#ccc"
              strokeDasharray="3 3"
              width={390}
              x={80}
              x1={80}
              x2={470}
              y={30}
              y1={206.25}
              y2={206.25}
            />
            <line
              fill="none"
              height={235}
              offset={
                Object {
                  "bottom": 35,
                  "brushBottom": 35,
                  "height": 235,
                  "left": 80,
                  "right": 30,
                  "top": 30,
                  "width": 390,
                }
              }
              stroke="#ccc"
              strokeDasharray="3 3"
              width={390}
              x={80}
              x1={80}
              x2={470}
              y={30}
              y1={147.5}
              y2={147.5}
            />
            <line
              fill="none"
              height={235}
              offset={
                Object {
                  "bottom": 35,
                  "brushBottom": 35,
                  "height": 235,
                  "left": 80,
                  "right": 30,
                  "top": 30,
                  "width": 390,
                }
              }
              stroke="#ccc"
              strokeDasharray="3 3"
              width={390}
              x={80}
              x1={80}
              x2={470}
              y={30}
              y1={88.75}
              y2={88.75}
            />
            <line
              fill="none"
              height={235}
              offset={
                Object {
                  "bottom": 35,
                  "brushBottom": 35,
                  "height": 235,
                  "left": 80,
                  "right": 30,
                  "top": 30,
                  "width": 390,
                }
              }
              stroke="#ccc"
              strokeDasharray="3 3"
              width={390}
              x={80}
              x1={80}
              x2={470}
              y={30}
              y1={30}
              y2={30}
            />
          </g>
          <g
            className="recharts-cartesian-grid-vertical"
          >
            <line
              fill="none"
              height={235}
              offset={
                Object {
                  "bottom": 35,
                  "brushBottom": 35,
                  "height": 235,
                  "left": 80,
                  "right": 30,
                  "top": 30,
                  "width": 390,
                }
              }
              stroke="#ccc"
              strokeDasharray="3 3"
              width={390}
              x={80}
              x1={177.5}
              x2={177.5}
              y={30}
              y1={30}
              y2={265}
            />
            <line
              fill="none"
              height={235}
              offset={
                Object {
                  "bottom": 35,
                  "brushBottom": 35,
                  "height": 235,
                  "left": 80,
                  "right": 30,
                  "top": 30,
                  "width": 390,
                }
              }
              stroke="#ccc"
              strokeDasharray="3 3"
              width={390}
              x={80}
              x1={372.5}
              x2={372.5}
              y={30}
              y1={30}
              y2={265}
            />
            <line
              fill="none"
              height={235}
              offset={
                Object {
                  "bottom": 35,
                  "brushBottom": 35,
                  "height": 235,
                  "left": 80,
                  "right": 30,
                  "top": 30,
                  "width": 390,
                }
              }
              stroke="#ccc"
              strokeDasharray="3 3"
              width={390}
              x={80}
              x1={80}
              x2={80}
              y={30}
              y1={30}
              y2={265}
            />
            <line
              fill="none"
              height={235}
              offset={
                Object {
                  "bottom": 35,
                  "brushBottom": 35,
                  "height": 235,
                  "left": 80,
                  "right": 30,
                  "top": 30,
                  "width": 390,
                }
              }
              stroke="#ccc"
              strokeDasharray="3 3"
              width={390}
              x={80}
              x1={470}
              x2={470}
              y={30}
              y1={30}
              y2={265}
            />
          </g>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
        >
          <line
            className="recharts-cartesian-axis-line"
            fill="none"
            height={30}
            orientation="bottom"
            stroke="#666"
            type="category"
            width={390}
            x={80}
            x1={80}
            x2={470}
            y={265}
            y1={265}
            y2={265}
          />
          <g
            className="recharts-cartesian-axis-ticks"
          >
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={30}
                orientation="bottom"
                stroke="#666"
                type="category"
                width={390}
                x={80}
                x1={177.5}
                x2={177.5}
                y={265}
                y1={271}
                y2={265}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height={30}
                orientation="bottom"
                stroke="none"
                textAnchor="middle"
                type="category"
                width={390}
                x={177.5}
                y={273}
              >
                <tspan
                  dy="0.71em"
                  x={177.5}
                >
                  0
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={30}
                orientation="bottom"
                stroke="#666"
                type="category"
                width={390}
                x={80}
                x1={372.5}
                x2={372.5}
                y={265}
                y1={271}
                y2={265}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height={30}
                orientation="bottom"
                stroke="none"
                textAnchor="middle"
                type="category"
                width={390}
                x={372.5}
                y={273}
              >
                <tspan
                  dy="0.71em"
                  x={372.5}
                >
                  auto
                </tspan>
              </text>
            </g>
          </g>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
        >
          <line
            className="recharts-cartesian-axis-line"
            fill="none"
            height={235}
            orientation="left"
            stroke="#666"
            type="number"
            width={60}
            x={20}
            x1={80}
            x2={80}
            y={30}
            y1={30}
            y2={265}
          />
          <g
            className="recharts-cartesian-axis-ticks"
          >
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={235}
                orientation="left"
                stroke="#666"
                type="number"
                width={60}
                x={20}
                x1={74}
                x2={80}
                y={30}
                y1={265}
                y2={265}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height={235}
                orientation="left"
                stroke="none"
                textAnchor="end"
                type="number"
                width={60}
                x={72}
                y={265}
              >
                <tspan
                  dy="0.355em"
                  x={72}
                >
                  0
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={235}
                orientation="left"
                stroke="#666"
                type="number"
                width={60}
                x={20}
                x1={74}
                x2={80}
                y={30}
                y1={206.25}
                y2={206.25}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height={235}
                orientation="left"
                stroke="none"
                textAnchor="end"
                type="number"
                width={60}
                x={72}
                y={206.25}
              >
                <tspan
                  dy="0.355em"
                  x={72}
                >
                  1
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={235}
                orientation="left"
                stroke="#666"
                type="number"
                width={60}
                x={20}
                x1={74}
                x2={80}
                y={30}
                y1={147.5}
                y2={147.5}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height={235}
                orientation="left"
                stroke="none"
                textAnchor="end"
                type="number"
                width={60}
                x={72}
                y={147.5}
              >
                <tspan
                  dy="0.355em"
                  x={72}
                >
                  2
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={235}
                orientation="left"
                stroke="#666"
                type="number"
                width={60}
                x={20}
                x1={74}
                x2={80}
                y={30}
                y1={88.75}
                y2={88.75}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height={235}
                orientation="left"
                stroke="none"
                textAnchor="end"
                type="number"
                width={60}
                x={72}
                y={88.75}
              >
                <tspan
                  dy="0.355em"
                  x={72}
                >
                  3
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={235}
                orientation="left"
                stroke="#666"
                type="number"
                width={60}
                x={20}
                x1={74}
                x2={80}
                y={30}
                y1={30}
                y2={30}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height={235}
                orientation="left"
                stroke="none"
                textAnchor="end"
                type="number"
                width={60}
                x={72}
                y={30}
              >
                <tspan
                  dy="0.355em"
                  x={72}
                >
                  4
                </tspan>
              </text>
            </g>
          </g>
        </g>
      </svg>
      <div
        className="recharts-legend-wrapper"
        style={
          Object {
            "height": "auto",
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 450,
          }
        }
      >
        <ul
          className="recharts-default-legend"
          style={
            Object {
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
            }
          }
        >
          <li
            className="recharts-legend-item legend-item-0"
            style={
              Object {
                "display": "inline-block",
                "marginRight": 10,
              }
            }
          >
            <svg
              className="recharts-surface"
              height={14}
              style={
                Object {
                  "display": "inline-block",
                  "marginRight": 4,
                  "verticalAlign": "middle",
                }
              }
              version="1.1"
              viewBox="0 0 32 32"
              width={14}
            >
              <path
                className="recharts-legend-icon"
                d="M0,4h32v24h-32z"
                fill="#003366"
                stroke="none"
              />
            </svg>
            <span
              className="recharts-legend-item-text"
              style={
                Object {
                  "color": "#003366",
                }
              }
            >
              Skill
            </span>
          </li>
          <li
            className="recharts-legend-item legend-item-1"
            style={
              Object {
                "display": "inline-block",
                "marginRight": 10,
              }
            }
          >
            <svg
              className="recharts-surface"
              height={14}
              style={
                Object {
                  "display": "inline-block",
                  "marginRight": 4,
                  "verticalAlign": "middle",
                }
              }
              version="1.1"
              viewBox="0 0 32 32"
              width={14}
            >
              <path
                className="recharts-legend-icon"
                d="M0,4h32v24h-32z"
                fill="#0099e6"
                stroke="none"
              />
            </svg>
            <span
              className="recharts-legend-item-text"
              style={
                Object {
                  "color": "#0099e6",
                }
              }
            >
              Exp
            </span>
          </li>
          <li
            className="recharts-legend-item legend-item-2"
            style={
              Object {
                "display": "inline-block",
                "marginRight": 10,
              }
            }
          >
            <svg
              className="recharts-surface"
              height={14}
              style={
                Object {
                  "display": "inline-block",
                  "marginRight": 4,
                  "verticalAlign": "middle",
                }
              }
              version="1.1"
              viewBox="0 0 32 32"
              width={14}
            >
              <path
                className="recharts-legend-icon"
                d="M0,4h32v24h-32z"
                fill="#ff6600"
                stroke="none"
              />
            </svg>
            <span
              className="recharts-legend-item-text"
              style={
                Object {
                  "color": "#ff6600",
                }
              }
            >
              Org
            </span>
          </li>
          <li
            className="recharts-legend-item legend-item-3"
            style={
              Object {
                "display": "inline-block",
                "marginRight": 10,
              }
            }
          >
            <svg
              className="recharts-surface"
              height={14}
              style={
                Object {
                  "display": "inline-block",
                  "marginRight": 4,
                  "verticalAlign": "middle",
                }
              }
              version="1.1"
              viewBox="0 0 32 32"
              width={14}
            >
              <path
                className="recharts-legend-icon"
                d="M0,4h32v24h-32z"
                fill="#ffa31a"
                stroke="none"
              />
            </svg>
            <span
              className="recharts-legend-item-text"
              style={
                Object {
                  "color": "#ffa31a",
                }
              }
            >
              JobSwitch
            </span>
          </li>
          <li
            className="recharts-legend-item legend-item-4"
            style={
              Object {
                "display": "inline-block",
                "marginRight": 10,
              }
            }
          >
            <svg
              className="recharts-surface"
              height={14}
              style={
                Object {
                  "display": "inline-block",
                  "marginRight": 4,
                  "verticalAlign": "middle",
                }
              }
              version="1.1"
              viewBox="0 0 32 32"
              width={14}
            >
              <path
                className="recharts-legend-icon"
                d="M0,4h32v24h-32z"
                fill="#cc33ff"
                stroke="none"
              />
            </svg>
            <span
              className="recharts-legend-item-text"
              style={
                Object {
                  "color": "#cc33ff",
                }
              }
            >
              Keyword
            </span>
          </li>
          <li
            className="recharts-legend-item legend-item-5"
            style={
              Object {
                "display": "inline-block",
                "marginRight": 10,
              }
            }
          >
            <svg
              className="recharts-surface"
              height={14}
              style={
                Object {
                  "display": "inline-block",
                  "marginRight": 4,
                  "verticalAlign": "middle",
                }
              }
              version="1.1"
              viewBox="0 0 32 32"
              width={14}
            >
              <path
                className="recharts-legend-icon"
                d="M0,4h32v24h-32z"
                fill="#66ff33"
                stroke="none"
              />
            </svg>
            <span
              className="recharts-legend-item-text"
              style={
                Object {
                  "color": "#66ff33",
                }
              }
            >
              Institute
            </span>
          </li>
        </ul>
      </div>
      <div
        className="recharts-tooltip-wrapper"
        style={
          Object {
            "MozTransform": "translate(undefinedpx, undefinedpx)",
            "OTransform": "translate(undefinedpx, undefinedpx)",
            "WebkitTransform": "translate(undefinedpx, undefinedpx)",
            "left": 0,
            "msTransform": "translate(undefinedpx, undefinedpx)",
            "pointerEvents": "none",
            "position": "absolute",
            "top": 0,
            "transform": "translate(undefinedpx, undefinedpx)",
            "visibility": "hidden",
          }
        }
      >
        <div
          className="recharts-default-tooltip"
          style={
            Object {
              "backgroundColor": "#fff",
              "border": "1px solid #ccc",
              "margin": 0,
              "padding": 10,
              "whiteSpace": "nowrap",
            }
          }
        >
          <p
            className="recharts-tooltip-label"
            style={
              Object {
                "margin": 0,
              }
            }
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "marginLeft": "80px",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row-reverse",
          "justifyContent": "flex-end",
        }
      }
    >
      <div
        style={
          Object {
            "marginBottom": "22px",
          }
        }
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "textAlign": "left",
            }
          }
        >
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "color": "#777",
                "fontSize": ".8rem",
                "lineHeight": "3rem",
                "padding": "0 0.2rem",
                "textTransform": "uppercase",
              }
            }
          />
        </div>
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column-reverse",
            "flexGrow": 0,
          }
        }
      >
        <div>
          <div
            style={
              Object {
                "display": "flex",
                "textAlign": "center",
              }
            }
          >
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "flexBasis": "3rem",
                  "flexGrow": "initial",
                  "flexShrink": 1,
                  "fontSize": ".8rem",
                  "overflow": "hidden",
                  "padding": "0.2rem 0",
                  "width": "3rem",
                }
              }
            >
              Skill
            </div>
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "flexBasis": "3rem",
                  "flexGrow": "initial",
                  "flexShrink": 1,
                  "fontSize": ".8rem",
                  "overflow": "hidden",
                  "padding": "0.2rem 0",
                  "width": "3rem",
                }
              }
            >
              Exp
            </div>
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "flexBasis": "3rem",
                  "flexGrow": "initial",
                  "flexShrink": 1,
                  "fontSize": ".8rem",
                  "overflow": "hidden",
                  "padding": "0.2rem 0",
                  "width": "3rem",
                }
              }
            >
              Org
            </div>
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "flexBasis": "3rem",
                  "flexGrow": "initial",
                  "flexShrink": 1,
                  "fontSize": ".8rem",
                  "overflow": "hidden",
                  "padding": "0.2rem 0",
                  "width": "3rem",
                }
              }
            >
              Job
            </div>
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "flexBasis": "3rem",
                  "flexGrow": "initial",
                  "flexShrink": 1,
                  "fontSize": ".8rem",
                  "overflow": "hidden",
                  "padding": "0.2rem 0",
                  "width": "3rem",
                }
              }
            >
              Key
            </div>
            <div
              style={
                Object {
                  "boxSizing": "border-box",
                  "flexBasis": "3rem",
                  "flexGrow": "initial",
                  "flexShrink": 1,
                  "fontSize": ".8rem",
                  "overflow": "hidden",
                  "padding": "0.2rem 0",
                  "width": "3rem",
                }
              }
            >
              Inst
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "flexGrow": 0,
            }
          }
        >
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "initial",
              }
            }
          >
            <div
              onClick={[Function]}
              style={
                Object {
                  "background": "rgb(0, 153, 255, NaN)",
                  "border": "1px solid #fff",
                  "borderRadius": "4px",
                  "borderWidth": "1px 1px 0 0",
                  "boxSizing": "border-box",
                  "color": "rgb(0, 0, 0)",
                  "cursor": "initial",
                  "flexBasis": "3rem",
                  "flexGrow": 0,
                  "flexShrink": 0,
                  "fontSize": ".8rem",
                  "height": "3rem",
                  "lineHeight": "3rem",
                  "marginTop": "5px",
                  "overflow": "hidden",
                  "textAlign": "center",
                }
              }
            >
              <div
                title="Pos(0, 0) = 0"
              >
                0
              </div>
            </div>
            <div
              onClick={[Function]}
              style={
                Object {
                  "background": "rgb(0, 153, 255, NaN)",
                  "border": "1px solid #fff",
                  "borderRadius": "4px",
                  "borderWidth": "1px 1px 0 0",
                  "boxSizing": "border-box",
                  "color": "rgb(0, 0, 0)",
                  "cursor": "initial",
                  "flexBasis": "3rem",
                  "flexGrow": 0,
                  "flexShrink": 0,
                  "fontSize": ".8rem",
                  "height": "3rem",
                  "lineHeight": "3rem",
                  "marginTop": "5px",
                  "overflow": "hidden",
                  "textAlign": "center",
                }
              }
            >
              <div
                title="Pos(0, 1) = 0"
              >
                0
              </div>
            </div>
            <div
              onClick={[Function]}
              style={
                Object {
                  "background": "rgb(0, 153, 255, NaN)",
                  "border": "1px solid #fff",
                  "borderRadius": "4px",
                  "borderWidth": "1px 1px 0 0",
                  "boxSizing": "border-box",
                  "color": "rgb(0, 0, 0)",
                  "cursor": "initial",
                  "flexBasis": "3rem",
                  "flexGrow": 0,
                  "flexShrink": 0,
                  "fontSize": ".8rem",
                  "height": "3rem",
                  "lineHeight": "3rem",
                  "marginTop": "5px",
                  "overflow": "hidden",
                  "textAlign": "center",
                }
              }
            >
              <div
                title="Pos(0, 2) = 0"
              >
                0
              </div>
            </div>
            <div
              onClick={[Function]}
              style={
                Object {
                  "background": "rgb(0, 153, 255, NaN)",
                  "border": "1px solid #fff",
                  "borderRadius": "4px",
                  "borderWidth": "1px 1px 0 0",
                  "boxSizing": "border-box",
                  "color": "rgb(0, 0, 0)",
                  "cursor": "initial",
                  "flexBasis": "3rem",
                  "flexGrow": 0,
                  "flexShrink": 0,
                  "fontSize": ".8rem",
                  "height": "3rem",
                  "lineHeight": "3rem",
                  "marginTop": "5px",
                  "overflow": "hidden",
                  "textAlign": "center",
                }
              }
            >
              <div
                title="Pos(0, 3) = 0"
              >
                0
              </div>
            </div>
            <div
              onClick={[Function]}
              style={
                Object {
                  "background": "rgb(0, 153, 255, NaN)",
                  "border": "1px solid #fff",
                  "borderRadius": "4px",
                  "borderWidth": "1px 1px 0 0",
                  "boxSizing": "border-box",
                  "color": "rgb(0, 0, 0)",
                  "cursor": "initial",
                  "flexBasis": "3rem",
                  "flexGrow": 0,
                  "flexShrink": 0,
                  "fontSize": ".8rem",
                  "height": "3rem",
                  "lineHeight": "3rem",
                  "marginTop": "5px",
                  "overflow": "hidden",
                  "textAlign": "center",
                }
              }
            >
              <div
                title="Pos(0, 4) = 0"
              >
                0
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "marginLeft": "50px",
        "marginRight": "50px",
      }
    }
  />,
  <div
    height="100%"
    style={
      Object {
        "marginBottom": "50px",
        "marginTop": "25px",
      }
    }
    width="100%"
  >
    <h3>
      Scatter Distribution of Applicants over Score
    </h3>
    <br />
    <div
      className="recharts-wrapper"
      style={
        Object {
          "cursor": "default",
          "height": 300,
          "position": "relative",
          "width": 500,
        }
      }
    >
      <svg
        className="recharts-surface"
        height={300}
        version="1.1"
        viewBox="0 0 500 300"
        width={500}
      >
        <defs>
          <clipPath
            id="recharts26-clip"
          >
            <rect
              height={230}
              width={400}
              x={80}
              y={20}
            />
          </clipPath>
        </defs>
        <g
          className="recharts-cartesian-grid"
        >
          <g
            className="recharts-cartesian-grid-horizontal"
          >
            <line
              fill="none"
              height={230}
              offset={
                Object {
                  "bottom": 50,
                  "brushBottom": 50,
                  "height": 230,
                  "left": 80,
                  "right": 20,
                  "top": 20,
                  "width": 400,
                }
              }
              stroke="#ccc"
              width={400}
              x={80}
              x1={80}
              x2={480}
              y={20}
              y1={20}
              y2={20}
            />
            <line
              fill="none"
              height={230}
              offset={
                Object {
                  "bottom": 50,
                  "brushBottom": 50,
                  "height": 230,
                  "left": 80,
                  "right": 20,
                  "top": 20,
                  "width": 400,
                }
              }
              stroke="#ccc"
              width={400}
              x={80}
              x1={80}
              x2={480}
              y={20}
              y1={250}
              y2={250}
            />
          </g>
          <g
            className="recharts-cartesian-grid-vertical"
          >
            <line
              fill="none"
              height={230}
              offset={
                Object {
                  "bottom": 50,
                  "brushBottom": 50,
                  "height": 230,
                  "left": 80,
                  "right": 20,
                  "top": 20,
                  "width": 400,
                }
              }
              stroke="#ccc"
              width={400}
              x={80}
              x1={80}
              x2={80}
              y={20}
              y1={20}
              y2={250}
            />
            <line
              fill="none"
              height={230}
              offset={
                Object {
                  "bottom": 50,
                  "brushBottom": 50,
                  "height": 230,
                  "left": 80,
                  "right": 20,
                  "top": 20,
                  "width": 400,
                }
              }
              stroke="#ccc"
              width={400}
              x={80}
              x1={480}
              x2={480}
              y={20}
              y1={20}
              y2={250}
            />
          </g>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
        >
          <line
            className="recharts-cartesian-axis-line"
            fill="none"
            height={30}
            name="Score"
            orientation="bottom"
            stroke="#666"
            type="number"
            width={400}
            x={80}
            x1={80}
            x2={480}
            y={250}
            y1={250}
            y2={250}
          />
          <g
            className="recharts-cartesian-axis-ticks"
          />
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
        >
          <line
            className="recharts-cartesian-axis-line"
            fill="none"
            height={230}
            name="No. of Applicants"
            orientation="left"
            stroke="#666"
            type="number"
            width={60}
            x={20}
            x1={80}
            x2={80}
            y={20}
            y1={20}
            y2={250}
          />
          <g
            className="recharts-cartesian-axis-ticks"
          />
        </g>
      </svg>
      <div
        className="recharts-tooltip-wrapper"
        style={
          Object {
            "MozTransform": "translate(undefinedpx, undefinedpx)",
            "OTransform": "translate(undefinedpx, undefinedpx)",
            "WebkitTransform": "translate(undefinedpx, undefinedpx)",
            "left": 0,
            "msTransform": "translate(undefinedpx, undefinedpx)",
            "pointerEvents": "none",
            "position": "absolute",
            "top": 0,
            "transform": "translate(undefinedpx, undefinedpx)",
            "visibility": "hidden",
          }
        }
      >
        <div
          className="recharts-default-tooltip"
          style={
            Object {
              "backgroundColor": "#fff",
              "border": "1px solid #ccc",
              "margin": 0,
              "padding": 10,
              "whiteSpace": "nowrap",
            }
          }
        >
          <p
            className="recharts-tooltip-label"
            style={
              Object {
                "margin": 0,
              }
            }
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "height": "100%",
        "marginBottom": "25px",
        "marginTop": "25px",
        "width": "100%",
      }
    }
  >
    <h5>
      No applicants applied!
    </h5>
  </div>,
]
`;
